lib/slack/middleware/get-installation.js

```diff
@@ -40,11 +40,13 @@ module.exports = async function getInstallation(req, res, next) {
     const installLink = `${protocol}://${host}/github/install/${owner.id}/${await state.stringify()}`;
 
     if (/api\.slack\.com/.test(req.headers['user-agent'])) {
-      res.json(new InstallGitHubApp(installLink));
-    } else {
-      res.redirect(installLink);
+      if (owner.id === parseInt(gitHubUser.id, 10)) {
+        return res.json(new InstallGitHubApp({ installLink }));
+      }
+      return res.json(new InstallGitHubApp({ installLink, ownerType: owner.type }));
```

Comment:
Since there's no different in behavior here, you pass owner to the InstallGitHubApp message and do this conditional there? That keeps this code focused on control logic, and the message focused on display logic.